migrations/v2.js

```diff
@@ -0,0 +1,37 @@
+import { describe , whereContent, whereFromPlugin, whereToPlugin, mutateContent, checkContent, updatePlugin } from 'adapt-migrations';
+
+function hasKey(object, key) {
+  if (!object) return false;
+  return Object.hasOwn(object, key);
+}
+
+function setObjectPathValue(object, path, value, force = false) {
+  if (!object) return;
+  const paths = path.split('.');
+  const key = paths.pop();
+  const target = paths.reduce((o, p) => {
+    if (!hasKey(o, p)) o[p] = {};
+    return o?.[p];
+  }, object);
+  if (!force && hasKey(target, key)) return;
+  target[key] = value;
+}
+
+describe('adapt-contrib-text - v2.0.0 to v2.0.1', async () => {
+  whereFromPlugin('adapt-contrib-text - from v2.0.0 to v2.0.1', { name: 'adapt-contrib-text', version: '<2.0.1'});
+  let course;
+  whereContent('adapt-contrib-text - where missing _globals ariaRegion', async content => {
+    course = content.find(({ _type }) => _type === 'course');
+    return !hasKey(course?._globals?._components?._text, 'ariaRegion');
+  });
+  mutateContent('adapt-contrib-text - add _globals ariaRegion', async () => {
+    setObjectPathValue(course, '_globals._components._text.ariaRegion', '');
```

Comment:
Again I'll not raise one for every instance but update setObjectPathValue to _.set

```suggestion
    _.set(course, '_globals._components._text.ariaRegion', '');
```

Reply:
If an author has changed the value from the default, we shouldn't apply the new default value. This would mean adding extra conditions before the set call, whereas `setObjectPathValue` takes care of this automatically:
https://github.com/adaptlearning/adapt-contrib-text/blob/7d253106bdbdfa58da9de08c4928f05e33baabc9/migrations/v6.js#L16

Would we rather add these extra conditions in and use lodash instead?
```JavaScript
if (!_.has(course, '_globals._components._text.ariaRegion')) _.set(course, '_globals._components._text.ariaRegion', '');
```

Switching to use the lodash method works in the v2 migration as the `whereContent` checks to see if the attribute is present first and returns early if so, but the [v6](https://github.com/adaptlearning/adapt-contrib-text/blob/7d253106bdbdfa58da9de08c4928f05e33baabc9/migrations/v6.js#L28) migration doesn't have this because it potentially needs to do the additional step of:
https://github.com/adaptlearning/adapt-contrib-text/blob/7d253106bdbdfa58da9de08c4928f05e33baabc9/migrations/v6.js#L29

Reply:
The idea is to only add helper functions that have behaviour which is easily described by its name.

https://github.com/adaptlearning/adapt-contrib-text/blob/7d253106bdbdfa58da9de08c4928f05e33baabc9/migrations/v6.js#L8-L18

`setObjectPathValue` uses `value` as a default value unless `force=true`. You would probably need to remove the `force` argument and have two functions, `setObjectPathDefaultValue` and `setObjectPathValue` in order to have self descriptive helper functions.

lodash is already installed and has got `setObjectPathValue` as `_.set` and `hasKey` as `_.has`, it just doesn't have `setObjectPathDefaultValue`. 

lodash also has many more useful helper functions and documentation, making it much easier and more useful and more widely beneficial for everyone to understand.

It is a little longer, but you can achieve a similar and simpler effect with lodash using:
```js
const ariaRegionPath = '_globals._components._text.ariaRegion'
if (!_.has(course, ariaRegionPath)) {
  _.set(course, ariaRegionPath, '');
}
```

I know @joe-allen-89 has had discussions with others about the other question `If an author has changed the value from the default, we shouldn't apply the new default value`. I'm sure he has some references or advice in that regard.